File: 0003-longest-substring-without-repeating-characters/0003-longest-substring-without-repeating-characters.java
class Solution {
    public int lengthOfLongestSubstring(String s) {
        
        int  i = 0, j = 0, max = 0;
        Set<Character> hs = new HashSet<Character>();
        int curr = 0;
        while(i <= j && j < s.length() &&(s.length()-i > max))
        {
            char ch = s.charAt(j);
            if(!hs.contains(ch))
            {
                hs.add(ch);
                max = Math.max(max,j - i + 1);
                j++;
            }
            else
            {
                hs.remove(s.charAt(i));
                i++;
            }
        }
        return max;
    }
}

File: 0003-longest-substring-without-repeating-characters/0003-longest-substring-without-repeating-characters.cpp
class Solution {
public:
    int lengthOfLongestSubstring(string s) {
        
        vector<int> vec(256,-1);
        int ans = 0, count = 0;
        int x = -1;
        for(int i = 0; i < s.length(); i++)
        {
            if(vec[s[i]] > (i-count-1))
            {
                count = i - (vec[s[i]])-1;
            }
            vec[s[i]] = i;
            count++;
            ans = max(ans, count);
        }
        return ans;
    }
};